### PR TITLE
doc/cephfs: edit front matter in client-auth.rst

### PR DESCRIPTION
**Pull request by zdover23**
Clarify the front matter in doc/cephfs/client-auth.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/cephfs/client-auth.rst [hunk 1]
```diff
@@ -2,15 +2,15 @@
 CephFS Client Capabilities
 ================================
 
-Use Ceph authentication capabilities to restrict your file system clients
-to the lowest possible level of authority needed.
+Ceph authentication capabilities are used to restrict file system clients to
```
Comment (anthonyeleven) [inline doc/cephfs/client-auth.rst:5]: CephFS

File: doc/cephfs/client-auth.rst [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
+the lowest level of authority necessary.
 
-.. note:: Path restriction and layout modification restriction are new features
-    in the Jewel release of Ceph.
+.. note:: Path restriction and layout-modification restriction were introduced
+   in the Jewel release of Ceph.
```
Comment (anthonyeleven) [inline doc/cephfs/client-auth.rst:9]: Maybe we don’t care about Infernalis and earlier?  Remove?

File: doc/cephfs/client-auth.rst [hunk 3]
```diff
@@ -2,15 +2,15 @@
...
 
-.. note:: Using Erasure Coded(EC) pools with CephFS is supported only with the
-   BlueStore Backend. They cannot be used as metadata pools and overwrites must
-   be enabled on the data pools.
+.. note:: Using Erasure Coded (EC) pools with CephFS is supported only with
+   :term:`BlueStore`. Erasure-coded pools cannot be used as metadata pools and
+   overwrites must be enabled on the data pools.
```
Comment (anthonyeleven) [inline doc/cephfs/client-auth.rst:13]: S/the//

**Review by anthonyeleven — APPROVED** [review]

File: doc/cephfs/client-auth.rst (hunk 3 above)
Comment (zdover23) [reply doc/cephfs/client-auth.rst:13]: I agree with this change, but I also think that this sentence should be split into two sentences. The way that it is now suggests that "overwrites" is the second item in a list that begins with "metadata pools", when in fact "overwrites" is the beginning of a second proposition, unrelated to the first.

File: doc/cephfs/client-auth.rst (hunk 2 above)
Comment (zdover23) [reply doc/cephfs/client-auth.rst:9]: I'm on the fence about removing stuff like this.

The case for removing information about when features were introduced: this is a technical manual, meant to explain how to use Ceph. This is not a historical record of the development of Ceph.

The case against removing information about when features were introduced: It's nice to know when features were introduced, and gives a sense of Ceph as an ongoing concern that responds to the world in which it serves the ends of its users and developers.

File: doc/cephfs/client-auth.rst (hunk 1 above)
Comment (zdover23) [reply doc/cephfs/client-auth.rst:5]: Good one.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/57121 - Squid backport
https://github.com/ceph/ceph/pull/57122 - Reef backport
https://github.com/ceph/ceph/pull/57123 - Quincy backport

**Pull request merged**